#!/usr/bin/env python
PACKAGE = "drive_ros_image_recognition"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Canny
gen.add("canny_param1", int_t, 0, "Canny parameter 1", 20, 10, 60)
gen.add("canny_param2", int_t, 0, "Canny parameter 2", 150, 50, 250)

# Hough
gen.add("hough_threshold", int_t, 0, "Hough vote threshold", 20, 3, 60)
gen.add("hough_min_line_length", int_t, 0, "Hough line length threshold", 10, 1, 40)
gen.add("hough_max_line_gap", int_t, 0, "Hough: maximum allowed gap to link points on same line", 5, 0, 200)

exit(gen.generate(PACKAGE, "drive_ros_image_recognition", "BarredAreaDetection"))

